### PR TITLE
feat(hip,aiter): route single-prefill through AITER mha_fwd (batch-mode) .so

### PR DESCRIPTION
**Pull request by demandal25**
## Summary

Single-prefill is batch=1; the natural fit is AITER's non-varlen `mha_fwd` template, whose CK Tile kernel instances are batch-mode and need no seqstart `[0, seqlen]` plumbing. PR #245 worked around the varlen-only routing by gating `is_group_mode=false` on ASM v3 eligibility — this PR removes that workaround by loading the right `.so` for each case.

#### C++ loader
- **`flashinfer/csrc_rocm/aiter_loader.cc`** — `build_so_name` parameterized with `include_logits` (the `mha_fwd` template has no `_logits/_nlogits` arm). Added `mha_fwd_variant_so_name`, `s_mf_*` cache, `get_aiter_mha_fwd_handle()` implementation. Renamed the existing varlen helpers to `mha_varlen_fwd_*` / `s_vl_*` / `get_aiter_mha_varlen_fwd_handle()`. Both loaders share `kMhaFwdSymbol` (both `.so` templates export the same dispatcher symbol — verified via `nm -D`).
- **`include/flashinfer/attention/aiter/aiter_loader.h`** — public API doc spells out the `has_logits_cap` footgun on `get_aiter_mha_fwd_handle`.

#### Dispatch
- **`include/flashinfer/attention/aiter/single_prefill.cuh`** — `has_logits_cap ? varlen_handle : fwd_handle` for loader selection; `is_group_mode = has_logits_cap` instead of `!use_asm_v3`. ASM v3 path unchanged (its eligibility predicate already excludes `has_logits_cap`).
- **`include/flashinfer/attention/aiter/batch_prefill.cuh`** — call site renamed to `get_aiter_mha_varlen_fwd_handle` (it's always been varlen).
- **`flashinfer/csrc_rocm/single_prefill_aiter.cu`** — `get_seqstart_ptrs` lookup gated on `params.logits_soft_cap > 0.0f` (the only case that still needs seqstart).

#### Python bootstrap
- **`flashinfer/prefill_rocm.py`** — renamed `_aiter_bootstrap_single_prefill` → `_aiter_bootstrap_single_prefill_varlen` for accuracy. Added `_aiter_bootstrap_single_prefill_mha_fwd(dtype, causal, has_lse, head_dim, device_idx)` that triggers `aiter.ops.mha.mha_fwd(...)`. The `mha_fwd` template ships no prebuilt `.so` files; every `(dtype, causal, has_lse)` is JIT-built on first use (~90 s per variant, cached on disk by AITER thereafter). Bootstrap call site collapsed into a single `if backend == "aiter":` block so auto-resolved-aiter and explicit-aiter share the path.

### Architecture / design notes

| Case | Loader used | `is_group_mode` | seqstart needed | Why |
|---|---|---|---|---|
| `logits_soft_cap == 0` (common) | `get_aiter_mha_fwd_handle` | `false` | no | Batch-mode CK + ASM v3 reachable via `use_asm_v3=true` |
| `logits_soft_cap > 0` | `get_aiter_mha_varlen_fwd_handle` | `true` | yes | `mha_fwd` template has no `_logits` arm |

The dispatched `aiter::mha_fwd` C++ symbol inside both `.so` files is the same dispatcher — it picks ASM v3 vs CK Tile per call based on `args.use_asm_v3`. The `.so` differs only in which CK Tile instances are baked in: `mha_fwd_*.so` has batch-mode instances, `mha_varlen_fwd_*.so` has group-mode instances. Loading the wrong `.so` would kernel-not-found inside `fmha_fwd_ck`; the predicate alignment between the loader selection and `is_group_mode` is the invariant that prevents this.

## Test plan

- [x] `pytest tests/rocm_tests/test_single_prefill_kernels_hip.py -k aiter --reruns 1` — **820 passed**, 1104 skipped (covers fp16 + bf16, hd ∈ {64, 128, 256}, causal/non-causal, LSE on/off, `logits_soft_cap ∈ {0.0, 8.0}` — both the new `mha_fwd` branch and the kept `mha_varlen_fwd` (logits-cap) branch are exercised)
- [x] `pytest tests/rocm_tests/test_single_prefill_kernels_hip.py --reruns 1` — **2456 passed**, 1392 skipped
- [x] `pytest tests/rocm_tests/ -k "prefill or aiter" --reruns 1` — **8575 passed**, 3528 skipped, 0 failed
- [x] `pre-commit run` on touched files

## Follow-ups (separate PRs, called out by /simplify)

- Per-key locking for `_aiter_bootstrap_*` so concurrent first-calls on different variants don't serialize through the single `_aiter_bootstrap_lock`.
- Shared helper to consolidate the four `_aiter_bootstrap_*` functions (~120 LOC → ~60).
- Move `_aiter_bootstrap_lock` + bootstrap helpers into `flashinfer/aiter_utils.py`.

### CODE REVIEW

File: flashinfer/csrc_rocm/aiter_loader.cc [hunk 1]
```diff
@@ -138,8 +157,19 @@ std::unordered_map<ExternCKey, void*, ExternCKeyHash> s_ec_cache;
 }  // namespace
 
 void* get_aiter_mha_fwd_handle(VariantKey const& key) {
-  std::string so_path = get_jit_dir() + "/" + variant_so_name(key);
-  return load_and_cache_sym(s_mu, s_cache, key, so_path, kMhaFwdSymbol, [&key, &so_path]() {
+  std::string so_path = get_jit_dir() + "/" + mha_fwd_variant_so_name(key);
```
[the remark this answers is not in the document]
Comment (demandal25) [reply flashinfer/csrc_rocm/aiter_loader.cc:166]: Fixed in ed6f0b4b. Added a runtime_error throw at the top of get_aiter_mha_fwd_handle when has_logits_cap=true. The current call site already gates on this, but the throw makes the constraint impossible to miss for future direct callers.

File: include/flashinfer/attention/aiter/single_prefill.cuh [hunk 2]
```diff
@@ -68,26 +64,31 @@ hipError_t SinglePrefillWithKVCacheDispatched(Params const& params, bool causal,
...
   args.how_v3_bf16_cvt = 0;
   args.data_type = dtype_str;
-  args.is_group_mode = !args.use_asm_v3;
-  // Defensive: group mode dereferences seqstart_*_ptr in AITER's CK Tile
-  // pipeline. Catch caller/callee predicate drift here rather than as a HIP
-  // memory fault inside the kernel.
-  if (args.is_group_mode && (cu_seqlens_q == nullptr || cu_seqlens_k == nullptr)) {
-    return hipErrorInvalidValue;
-  }
+  args.is_group_mode = has_logits_cap;
```
[the remark this answers is not in the document]
Comment (demandal25) [reply include/flashinfer/attention/aiter/single_prefill.cuh:91]: Good catch — fixed in ed6f0b4b. Added test_single_prefill_aiter_bf16 (216 parameterized cases: qo_len × kv_len × num_qo_heads × hd ∈ {64,128,256} × causal × return_lse). Exercises the bf16 ASM v3 fast path on hd128 plus the bf16 CK Tile fallback on hd64/hd256, all under logits_soft_cap=0 (the mha_fwd .so path). warmup_jit fixture extended to include bf16. 192 pass (24 skipped for causal+qo>kv).
